### ISSUE TITLE
Incremental compilation warning when precompiling Gen

### ISSUE THREAD

**Issue by ztangent** [probcomp/Gen.jl#292]
These warnings have been occurring during precompilation of `Gen` (or packages that use `Gen`), suggesting that incremental compilation may be broken:

```
WARNING: Method definition special_reverse_exec!(ReverseDiff.SpecialInstruction{typeof(Base.fill), I, O, C} where C where O where I) in module ReverseDiff at [...]\.julia\packages\ReverseDiff\tCOFA\src\macros.jl:213 overwritten in module Gen at [...]\.julia\packages\Gen\5JiNL\src\backprop.jl:35.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition special_forward_exec!(ReverseDiff.SpecialInstruction{typeof(Base.fill), I, O, C} where C where O where I) in module ReverseDiff at [...]\.julia\packages\ReverseDiff\tCOFA\src\macros.jl:229 overwritten in module Gen at [...]\.julia\packages\Gen\5JiNL\src\backprop.jl:44.
  ** incremental compilation may be fatally broken for this module **
```

According to @marcoct this looks like it's because "someone added support for `fill` to ReverseDiff, and that Gen had this built-in for a while." The fix should be just to remove those definitions of `fill` from Gen.

**Comment by bzinberg** [comment]
might want to confirm that the definitions in ReverseDiff.jl are equivalent to the ones in Gen?  (maybe not worth it idk)

**Comment by ztangent** [comment]
@bzinberg I looked at the ReverseDiff.jl implementation and it looks like it's doing the same thing, they just auto-define the `special_forward_exec!` and `special_reverse_exec!` methods for `fill` within their `@grad` macro. Just made the fix in #293.
